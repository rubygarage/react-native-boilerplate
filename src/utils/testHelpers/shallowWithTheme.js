import React from 'react';
import { ThemeProvider } from 'styled-components/native';
// eslint-disable-next-line import/no-extraneous-dependencies
import { shallow } from 'enzyme';

import mockedTheme from './mockedTheme';

export default (tree) => shallow(
  <ThemeProvider theme={mockedTheme}>
    {tree}
  </ThemeProvider>,
);
