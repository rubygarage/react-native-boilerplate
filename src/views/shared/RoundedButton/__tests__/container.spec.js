import React from 'react';

import shallowWithTheme from 'utils/testHelpers/shallowWithTheme';
import diveTo from 'utils/testHelpers/diveToEnzyme';
import RoundedButtonWrapped, { RoundedButtonContainer } from '../container';

const defaultProps = { iconName: 'video-off', isActive: true, onPress: jest.fn() };

describe('RoundedButton container', () => {
  it('renders correctly default', () => {
    const wrapper = shallowWithTheme(<RoundedButtonWrapped {...defaultProps} />);
    const container = diveTo(wrapper, RoundedButtonContainer);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with active icon', () => {
    const props = { ...defaultProps, iconNameActive: 'video' };
    const wrapper = shallowWithTheme(<RoundedButtonWrapped {...props} />);
    const container = diveTo(wrapper, RoundedButtonContainer);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly inactive', () => {
    const props = { ...defaultProps, iconNameActive: 'video', isActive: false };
    const wrapper = shallowWithTheme(<RoundedButtonWrapped {...props} />);
    const container = diveTo(wrapper, RoundedButtonContainer);
    expect(container).toMatchSnapshot();
  });
});
