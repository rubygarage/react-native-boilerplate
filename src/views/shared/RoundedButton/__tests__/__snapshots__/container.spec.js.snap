// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton container renders correctly default 1`] = `
<Styled(TouchableOpacity)
  isActive={true}
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    color="[mocked white]"
    name="video-off"
    size={28}
  />
</Styled(TouchableOpacity)>
`;

exports[`RoundedButton container renders correctly inactive 1`] = `
<Styled(TouchableOpacity)
  isActive={false}
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    color="[mocked grey700]"
    name="video-off"
    size={28}
  />
</Styled(TouchableOpacity)>
`;

exports[`RoundedButton container renders correctly with active icon 1`] = `
<Styled(TouchableOpacity)
  isActive={true}
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    color="[mocked white]"
    name="video"
    size={28}
  />
</Styled(TouchableOpacity)>
`;
